docs/configuration/http_filters/fault_filter.rst

```diff
@@ -66,6 +67,12 @@ upstream_cluster:
   the config are present in the request with the same values (or based on presence if the ``value``
   field is not in the config).
 
+downstream_nodes:
+  *(optional, array)* Faults are injected for the specified list of downstream hosts. If this setting is
+  not set, faults are injected for all downstream nodes. Downstream node name is taken from
+  :ref:`the HTTP x-envoy-downstream-service-node <config_http_conn_man_headers_downstream-service-node>`
+  header and compared against downstream_nodes list.
+
```

Comment:
That is perfectly fine. But what I was driving at was, do we need to add a config element for this? 
Why can't this be done in the followign way:
```
fault_filter1:
  headers:
    x-envoy-downstream-service-node: canary1
  delay:
    duration_ms: 1000
fault_filter2:
  headers:
    x-envoy-downstream-service-node: canary2
  delay:
    duration_ms: 1000
```

The logic from lines 96-110 in fault_filter.cc can stay as it is [the one that checks if the downstream cluster name in the header is present as part of the runtime with overrides for delay and abort]

Make sense? Essentially piggyback on the header match itself, instead of adding a new config element. If there is a header match for X-Envoy-downstream-cluster, then pull it out, check with runtime, etc.